server/src/main/java/org/elasticsearch/index/mapper/SourceLoader.java

```diff
@@ -87,8 +106,8 @@ class Synthetic implements SourceLoader {
         private final Set<String> requiredStoredFields;
         private final SourceFieldMetrics metrics;
 
-        public Synthetic(Mapping mapping, SourceFieldMetrics metrics) {
-            this.syntheticFieldLoaderLeafSupplier = mapping::syntheticFieldLoader;
+        public Synthetic(Supplier<SyntheticFieldLoader> fieldLoaderSupplier, SourceFieldMetrics metrics) {
```

Comment:
Nit: add a short comment about the supplier arg.

Reply:
Provided in https://github.com/elastic/elasticsearch/pull/109472/commits/e01a6011d2c14862163c0da7adb924ca9a925249